### PR TITLE
acid-block | Replace pkg_resources to importlib_resources | Drop support for python 3.8

### PR DESCRIPTION
**Pull request by ttqureshi**
# Overview
This PR:
- replaces `pkg_resources` api from `importlib_resources` api
- drops support for `python 3.8` and add support for `python 3.11`
---
Tickets:
- [Move on from deprecated pkg_resources api](https://github.com/openedx/acid-block/issues/179)
- [Drop Python 3.8 & Add Support for Python 3.12](https://github.com/openedx/acid-block/issues/184)

Followed [this](https://importlib-resources.readthedocs.io/en/latest/migration.html) migration guide.

## Testing
I have tested the changes against the following steps:
- Install the `acid-block` to `xblock-sdk`.
- Functional testing

### `XBlock Acid single block test`
image

### `XBlock Acid Parent test`
image

### CODE REVIEW

File: acid/acid.py [hunk 1]
```diff
@@ -12,8 +12,10 @@
 
 try:
     from web_fragments.fragment import Fragment
+    from xblock.utils.resources import ResourceLoader
```
Comment (farhan) [inline acid/acid.py:15]: I think it should be in separate try catch

**Review by farhan — CHANGES_REQUESTED** [review]
1 change

**Review by feanil — APPROVED** [review]
Please bump the version so we can also do a new release.

**Comment by ttqureshi** [comment]
> Please bump the version so we can also do a new release.

ok 👍

**Review by farhan — APPROVED** [review]

**Pull request merged**